src/models/logic/transferAsset.ts

```diff
@@ -253,5 +247,19 @@ export async function createTransferAsset(
             createAssetTypeLog(transaction, assetType, options)
         )
     );
+    await Promise.all(
+        addresses.map(address =>
+            Promise.all(
+                assetTypes.map(assetType =>
+                    createAssetAddressLog(
+                        transaction,
+                        address,
+                        assetType,
+                        options
+                    )
+                )
+            )
+        )
+    );
```

Comment:
Am I right? I understand the cause of the code like this.
A transaction can have many inputs and outputs. It is hard to track which input is related to which output. So we add all the combinations in the AssetAddressLog table instead.

Reply:
Basically, yes. I see there is no clear way to tell which addresses are not related to a specific asset type in a point of CodeChain transaction's view. So we consider that every address is related to every asset type in the transaction.

Reply:
Ok, I will add a comment.

Reply:
@ScarletBlue 
```
    // NOTE: TransferAsset can have many inputs and outputs, and some addresses
    // may not related to a specific asset type when it's a kind of
    // CoinJoin(https://en.bitcoin.it/wiki/CoinJoin) transaction. However, it
    // creates all the combinations of the asset addresses and the asset types
    // because there is no clear way to tell which addresses are not related to
    // which asset types
```